### ISSUE TITLE
Sendtoroute syntax is unclear

### ISSUE THREAD

**Issue by ZapUser77** [lightningnetwork/lnd#1602]
### Background

Trying to use sendtoroute command to rebalance channels

### Your environment

* version of `lnd`
0.4.2
* which operating system (`uname -a` on *Nix)
* version of `btcd`, `bitcoind`, or other backend
Bitcoind
* any other relevant environment details

### Steps to reproduce

Sendtoroute

Can't use queryroutes to self-balance.

using the --routes parameter to manually specify a JSON encoded set of routes in the format of the return value of queryroutes:
           (lncli sendtoroute --payment_hash=<pay_hash> --routes=<route>)

passing the routes as a positional argument:
           (lncli sendtoroute --payment_hash=pay_hash <route>)

This is unclear what the <route> syntax should look like.  I found a route to the next to final hop, along with the channel id (from my side... to both nodes number a channel the same number?).

Do I just create a file?:

Routes
[
        {
            "total_time_lock": 533082,
            "total_fees": "0",
            "total_amt": "900000",
            "hops": [
                {
                    "chan_id": "<Outgoing Channel>",
                    "chan_capacity": "988914",
                    "amt_to_forward": "900000",
                    "fee": "0",
                    "expiry": 533068,
                    "amt_to_forward_msat": "900000000",
                    "fee_msat": "0"
                },
                {
                    "chan_id": "<Intermediate Channel>",
                    "chan_capacity": "2000000",
                    "amt_to_forward": "900000",
                    "fee": "2251",
                    "expiry": 533068,
                    "amt_to_forward_msat": "900000000",
                    "fee_msat": "2251"
                }
#Next part was calculated from getchannelinfo <Incoming Channel>  As was Fee and fee msat and Total_time_lock
               {
                    "chan_id": "<Incoming Channel>",
                    "chan_capacity": "990000",
                    "amt_to_forward": "900000",
                    "fee": "0",
                    "expiry": 533068,
                    "amt_to_forward_msat": "900000000",
                    "fee_msat": "0"
                }
            ],
            "total_fees_msat": "0",
            "total_amt_msat": "900000000"
        },
]

When trying to add in the last hop (my node):

  "node2_policy": {
        "time_lock_delta": 144,
        "min_htlc": "1000",
        "fee_base_msat": "1000",
        "fee_rate_milli_msat": "2500"

Give the above example, can you provide *exactly* how to use the "Sendtoroute" command to properly execute?

**Comment by ZapUser77** [comment]
-_-   Why did you close the ticket?  This is still exactly as unclear as when I posted it.  

I understand that you believe you're providing useful information.  But, you're providing what's I already found in the commands "help"... which was less than useless to someone who doesn't know how to find the format of the "queryroutes JSon".

I removed the channel ID's for security reasons.  And the above is exactly the output of of query routes.

Ok.

Let me ask it another way.  How do I find out what the "Queryroutes **JSON**" format looks like?

**Comment by ZapUser77** [comment]
I'm trying to find the syntax.
Do I create a file?
Do I just do an excessively long command?  If so, what is the specific syntax?  How do I find the syntax of "queryroutes json".

**Comment by Roasbeef** [comment]
```
NAME:
   lncli sendtoroute - send a payment over a predefined route

USAGE:
   lncli sendtoroute [command options] [arguments...]

DESCRIPTION:
   
  Send a payment over Lightning using a specific route. One must specify
  a list of routes to attempt and the payment hash. This command can even
  be chained with the response to queryroutes. This command can be used
  to implement channel rebalancing by crafting a self-route, or even
  atomic swaps using a self-route that crosses multiple chains.

  There are three ways to specify routes:
     * using the --routes parameter to manually specify a JSON encoded
       set of routes in the format of the return value of queryroutes:
           (lncli sendtoroute --payment_hash=<pay_hash> --routes=<route>)

     * passing the routes as a positional argument:
           (lncli sendtoroute --payment_hash=pay_hash <route>)

     * or reading in the routes from stdin, which can allow chaining the
       response from queryroutes, or even read in a file with a set of
       pre-computed routes:
           (lncli queryroutes --args.. | lncli sendtoroute --payment_hash= -

       notice the '-' at the end, which signals that lncli should read
       the route in from stdin
  

OPTIONS:
   --payment_hash value, --pay_hash value  the hash to use within the payment's HTLC
   --routes value, -r value                a json array string in the format of the response of queryroutes that denotes which routes to use
   ``

**Comment by Roasbeef** [comment]
You can use `queryroutes` to see what the syntax looks like. If you look above, there're three options: pass as a keyword arg, pass positionally, or read in from stdin.

**Comment by ZapUser77** [comment]
-_- 
I quoted the help in the original ticket... and said it was insufficient.
You quote the help... and tell me is.

I quoted the output of queryroutes.

I still have no idea what the syntax should be.  

Can you *please* just give me the an example of what the command should be?

**Comment by Roasbeef** [comment]
If you don't know what JSON is, then I'm afraid I can't help you out here. I recommend you just experiment with the command and work it out. The issue tracker is for actual bug or requests related to the codebase, we can't walk you through from JSON to rebalancing.

**Comment by ZapUser77** [comment]
Ok, Just tell me how to locate the format?  I've issued the command "Queryroutes".
Are you saying just use the output?
I don't need you to walk me through anything.  I'm just need *basic* syntax example... something that should be IN the help of the command itself.

**Comment by Roasbeef** [comment]
Yes, that is the output. The help statement explicitly denotes that the expected output.

**Comment by ZapUser77** [comment]
Why not just Say use the "queryroutes output syntax"?  I was looking for a .json file.

"The help statement explicitly denotes that the expected output." Only to those who are familiar with coding languages that use JSON.  Not every programmer is. Which is the entire point.

The point of the ticket is:
Please make it clear in the command help what the syntax should be.

**Comment by Roasbeef** [comment]
> I'd recommend first just using the output of queryroutes
